### PR TITLE
ENH: Add support for reading string lists

### LINKED ISSUE #1140: BUG: GeoJSON driver returning StringList for a string containing a list
Code to create the file:
```python
import fiona

schema = {
    "properties": {
        "time_range": "str",
    },
    "geometry": "Point",
}
with fiona.open(
    "fio_test.geojson", "w", driver="GeoJSON", schema=schema, crs="EPSG:4326"
) as fds:
    fds.writerecords(
        [
            {
                "id": 1,
                "geometry": {"type": "Point", "coordinates": [0.0, 0.0]},
                "properties": {
                    "time_range": '["2020-01-01", "2020-01-02"]',
                },
            }
        ]
    )
```
file contents:
```json
{
"type": "FeatureCollection",
"crs": { "type": "name", "properties": { "name": "urn:ogc:def:crs:OGC:1.3:CRS84" } },
"features": [
{ "type": "Feature", "properties": { "time_range": [ "2020-01-01", "2020-01-02" ] }, "geometry": { "type": "Point", "coordinates": [ 0.0, 0.0 ] } }
]
}
```
```bash
ogrinfo -so fio_test.geojson fio_test
```
```
...
time_range: StringList (0.0)
```
```bash
fio info fio_test.geojson
```
```
WARNING:fiona.ogrext:Skipping field time_range: invalid type 5
{"driver": "GeoJSON", "schema": {"properties": {}, ...
```

It is strange that the original type was a string and it was converted to a StringList type.
Also helpful to note that this is not an issue with the GPKG driver.

## version information
```
Fiona version: 1.8.21
GDAL version: 3.5.1
PROJ version: 9.0.1

OS: Linux 5.15.0-47-generic
Python: 3.10.5
```

### PR DESCRIPTION
**Pull request by snowman2**
Closes #1140
```
fio info fio_test.geojson
{"driver": "GeoJSON", "schema": {"properties": {"time_range": "List[str]"}, ....
```

### CODE REVIEW

**Comment by snowman2** [comment]
The rstcheck failure is unrelated to the changes in this PR.

**Comment by snowman2** [comment]
@sgillies just remembered this PR. Do you think that it is a small enough change to be a candidate for something to include in 1.9.

File: fiona/schema.pyx [hunk 1]
```diff
@@ -24,7 +26,7 @@ FIELD_TYPES = [
     'float',        # OFTReal, Double Precision floating point
     None,           # OFTRealList, List of doubles
     'str',          # OFTString, String of UTF-8 chars
-    None,           # OFTStringList, Array of strings
+    'List[str]',    # OFTStringList, Array of strings
```
Comment (sgillies) [inline fiona/schema.pyx:29]: @snowman2 yes, I agree this is the right way to do it. Thank you!

**Review by sgillies — APPROVED** [review]

**Comment by snowman2** [comment]
Thanks @sgillies :+1:

**Pull request merged**